:bug: [FEAT] 프로필 이미지 S3 URL 조합 처리 로직 추가

hyosikkk opened the pull request:
🔸 UserInfoCard.tsx
- profileImageUrl가 S3 key인 경우를 고려해 `S3_BASE_URL`과 조합해서 실제 이미지 URL 구성
- S3 key가 이미 http로 시작하는 전체 URL인 경우는 그대로 사용

🔸 ProfileSummaryCard.tsx
- `user.profileImageUrl` 값이 S3 key일 경우를 대비해 프론트에서 S3 BASE URL과 조합 처리
- 백엔드에서 전체 URL을 내려주지 않더라도 프론트에서 안전하게 처리 가능하도록 수정

hyosikkk committed:
:bug: [FEAT] 프로필 이미지 S3 URL 조합 처리 로직 추가

🔸 UserInfoCard.tsx
- profileImageUrl가 S3 key인 경우를 고려해 `S3_BASE_URL`과 조합해서 실제 이미지 URL 구성
- S3 key가 이미 http로 시작하는 전체 URL인 경우는 그대로 사용

🔸 ProfileSummaryCard.tsx
- `user.profileImageUrl` 값이 S3 key일 경우를 대비해 프론트에서 S3 BASE URL과 조합 처리
- 백엔드에서 전체 URL을 내려주지 않더라도 프론트에서 안전하게 처리 가능하도록 수정

hyosikkk committed:
:bug: [Feat]GitHub OAuthCallback에서 JSON 응답 여부 체크 로직 추가

- content-type이 application/json인지 확인 후만 user.json() 파싱
- HTML 등 비정상 응답에 대한 예외 처리 강화

Kojaewoong0504 commented on src/pages/home/components/ProfileSummaryCard.tsx:
```diff
@@ -22,12 +22,15 @@ const ProfileSummaryCard = ({ user, onViewProfile }: Props) => {
   const winRate = user.win_rate != null ? user.win_rate.toFixed(2) : "0.00";
 
+  const S3_BASE_URL = "https://codeground-profile.s3.ap-northeast-2.amazonaws.com";
+  const profileImageUrl = user.profileImageUrl ? `${S3_BASE_URL}/${user.profileImageUrl}` : undefined;
```
S3_BASE_URL이 하드 코딩 되어 있는데 승인 하겠지만 이건 추 후에 환경 변수에 넣던지 해야 할 것 같아요

Kojaewoong0504 commented on src/pages/profile/components/UserInfoCard.tsx:
```diff
@@ -4,34 +4,40 @@ import { User } from "lucide-react";
-  </CyberCard>
-);
+const UserInfoCard = ({ nickname, profileImageUrl: rawUrl, rank, tier, lp, onEdit }: Props) => {
+  const S3_BASE_URL = "https://codeground-profile.s3.ap-northeast-2.amazonaws.com";
```
여기도 하드코딩 된 내용이 있고 이거 중복 된 것 같아요

hyosikkk replied:
여러 컴포넌트(ProfileSummaryCard.tsx, OAuthCallback.tsx, UserInfoCard.tsx)에 하드코딩되어 있던 S3 기본 URL을 .env 환경 변수로 분리하고,
공통 유틸리티 함수 getAbsoluteUrl을 추가하여 이미지 URL 생성 로직을 일원화했습니다.
중복 코드를 제거하고 유지보수성과 일관성을 개선했습니다.

Kojaewoong0504 reviewed (approved):
승인 합니다.

다음에 올릴 때는 하드코딩 된 내용 처리 부탁합니다.

Mugen-Houyou commented on src/pages/login/components/OAuthCallback.tsx:
```diff
@@ -1,53 +1,56 @@
+            } catch (error) {
+                console.error("OAuth 처리 중 예외:", error);
+                navigate(
+                    `/login?error=network_error&message=OAuth 처리 중 오류가 발생했습니다.`
```
`/login?error=network_error&message=OAuth 처리 중 오류가 발생했습니다.`
 이거 말인데, 이렇게 해도 URL 파싱 잘되나요? (잘 몰라서)

hyosikkk replied:
navigate("/login?error=...&message=...")는 브라우저가 자동으로 한글 같은 특수문자도 인코딩해주고, 이후 /login 페이지에서 URLSearchParams로 location.search 파싱하면 "error", "message" 다 정상적으로 가져올 수 있습니다 추가적인 인코딩/디코딩 처리 따로 안 해도 된다고 합니다 !

hyosikkk committed:
:recycle: [REFACTOR] S3 URL 환경 변수 관리 및 중복 코드 제거

   - 여러 컴포넌트에 하드코딩된 S3 기본 URL을 .env 환경 변수(VITE_S3_BASE_URL)로 분리
   - 중복되는 URL 생성 로직을 getAbsoluteUrl 유틸리티 함수로 통합하여 재사용성 향상
   - ProfileSummaryCard, OAuthCallback, UserInfoCard 컴포넌트에 신규 유틸리티 함수 적용
   - 실수로 변경되었던 .env.production의 VITE_API_URL 값을 원상 복

Kojaewoong0504 reviewed (approved):
가보자고

hmmwtf reviewed (approved):
성불 슛!
